benchmarks/randomized_benchmarking.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+import functools
+import numpy as np
+import cirq
+from cirq.experiments.qubit_characterizations import _single_qubit_cliffords, _find_inv_matrix
+
+
+def dot(*args: np.ndarray) -> np.ndarray:
+    return functools.reduce(np.dot, args)
+
+
+class SingleQubitRandomizedBenchmarking:
+    """Benchmarks circuit construction time for single qubit randomized benchmarking circuits.
+
+    Given a combination of `depth`, `num_qubits` and `num_circuits`, the benchmark constructs
+    `num_circuits` different circuits, each spanning `num_qubits` and containing `depth` moments.
+    Each moment of the circuit contains a single qubit clifford operation for each qubit.
+
+    Thus, the generated circuits have `depth * num_qubits` single qubit clifford operations.
+    """
+
+    params = [[1, 10, 50, 100, 250, 500, 1000], [100], [20]]
+    param_names = ["depth", "num_qubits", "num_circuits"]
+    timeout = 600  # Change timeout to 10 minutes instead of default 60 seconds.
+
+    def setup(self, *_):
+        self.sq_xz_matrices = np.array(
+            [dot(*[cirq.unitary(c) for c in group]) for group in _single_qubit_cliffords().c1_in_xz]
```

Comment:
Need to multiply matrices in the reverse order of how the gates appear in "circuit order".
```suggestion
            [dot(*(cirq.unitary(c) for c in reversed(group))) for group in _single_qubit_cliffords().c1_in_xz]
```